apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/internal/SchemaHelper.kt

```diff
@@ -0,0 +1,60 @@
+package com.apollographql.apollo.gradle.internal
+
+import com.apollographql.apollo.compiler.toJson
+import com.squareup.moshi.JsonWriter
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import okio.buffer
+import okio.sink
+import java.io.ByteArrayOutputStream
+import java.util.concurrent.TimeUnit
+
+internal object SchemaHelper {
+  private fun newOkHttpClient(): OkHttpClient {
+    val connectTimeoutSeconds = System.getProperty("okHttp.connectTimeout", "600").toLong()
```

Comment:
it may be worth having some kind of automatic retry on disconnect, but 🤷

Reply:
I'll keep it simple for now. Given that it's mostly used from the cli, it shouldn't be too hard to relaunch the command. But if disconnects happen too often, it shouldn't be too hard to do.